src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -3531,69 +3537,63 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         return _core.SelectedText(trimTrailingWhitespace);
     }
 
-    // Method Description:
-    // - Triggers an update on scrollbar to redraw search scroll marks.
-    // - Called when the search results have changed, and the scroll marks'
-    //   positions need to be updated.
-    void TermControl::_UpdateSearchScrollMarks()
+    void TermControl::_refreshSearch()
     {
-        // Manually send a scrollbar update, on the UI thread. We're already
-        // UI-driven, so that's okay. We're not really changing the scrollbar,
-        // but we do want to update the position of any search marks. The Core
-        // might send a scrollbar update event too, but if the first search hit
-        // is in the visible viewport, then the pips won't display until the
-        // user first scrolls.
-        auto scrollBar = ScrollBar();
-        ScrollBarUpdate update{
-            .newValue = scrollBar.Value(),
-            .newMaximum = scrollBar.Maximum(),
-            .newMinimum = scrollBar.Minimum(),
-            .newViewportSize = scrollBar.ViewportSize(),
-        };
-        _throttledUpdateScrollbar(update);
+        if (!_searchBox || _searchBox->Visibility() != Visibility::Visible)

```

Comment:
Could this cause a problem when the search box is closed and the user uses Find previous/next match actions to navigate through the results?

Reply:
Opened the search box, and typed "tusha" into it, and then closed it. Search highlights were removed as expected. Used Find next search match, and the highlights are active again. Entered some text containing "tusha". Highlight didn't refresh. After running `clear` command:

![image](https://github.com/microsoft/terminal/assets/55626797/ad1791be-fd7b-47d7-a44c-98cef08a27e7)

Reply:
(Did we fix this?)

Reply:
Both underlying issues should be fixed now: When using the action, we now open the search box which triggers a search once opened, and while searching we always call `TriggerSearchHighlight` now.